src/hardware_functions.cpp

```diff
@@ -3,28 +3,42 @@
 #include <QDebug>
 #include <QtGlobal>
 
-#if QT_VERSION < QT_VERSION_CHECK(5, 11, 0)
-
 #if (defined(_WIN32) || defined(_WIN64))
 #include <windows.h>
-
-static DWORD dwVolSerial;
-static BOOL bIsRetrieved;
+#include <sddl.h>
 
 QString get_hdid()
 {
-  bIsRetrieved = GetVolumeInformation(TEXT("C:\\"), nullptr, 0, &dwVolSerial,
-                                      nullptr, nullptr, nullptr, 0);
-
-  if (bIsRetrieved)
-    return QString::number(dwVolSerial, 16);
-  else
-    // a totally random string
-    // what could possibly go wrong
-    return "gxsps32sa9fnwic92mfbs0";
-}
+    HANDLE hToken;
+    HANDLE pHandle;
+    PTOKEN_USER pToken;
+    DWORD uSize = 0;
+    LPWSTR HDIDParam;
+
+    pHandle = GetCurrentProcess();
+    OpenProcessToken(pHandle, TOKEN_QUERY, &hToken);
+    if (!GetTokenInformation(hToken, (TOKEN_INFORMATION_CLASS)1, NULL, 0, &uSize))
+    {
+        if (GetLastError() != ERROR_INSUFFICIENT_BUFFER) {
+            return "gxsps32sa9fnwic92mfbs1";
+        }
+    }
 
-#elif (defined(LINUX) || defined(__linux__))
+    pToken = (PTOKEN_USER)GlobalAlloc(GPTR, uSize);
+
+    if (!GetTokenInformation(hToken, (TOKEN_INFORMATION_CLASS)1, pToken, uSize, &uSize))
+    {
+        return "gxsps32sa9fnwic92mfbs2";
+    }
+
+    ConvertSidToStringSidW(pToken->User.Sid, &HDIDParam);
+    QString returnHDID = QString::fromWCharArray(HDIDParam);
+    CloseHandle(hToken);
+    CloseHandle(pHandle);
```

Comment:
```suggestion
```
you can safely remove this line as GetCurrentProcess() will always return a pseudohandle which cannot be closed

> The pseudo handle need not be closed when it is no longer needed. Calling the [CloseHandle](https://help.intervalzero.com/product_help/RTX64_4/Content/Topics/SDK/Windows/CloseHandle.htm) function with a pseudo handle has no effect.
https://help.intervalzero.com/product_help/RTX64_4/Content/Topics/SDK/Windows/GetCurrentProcess.htm